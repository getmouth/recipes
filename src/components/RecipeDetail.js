import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import classNames from 'classnames';

const RecipeDetail = props => {
  if (!props.recipe) {
    return (
      <p
        className={classNames('h3 p2 bg-white italic center', props.className)}
        style={props.style}
      >
        Please select a recipe to view the details
      </p>
    );
  }
  return (
    <div
      className={classNames('p2 bg-white', props.className)}
      style={props.style}
    >
      <h2 className="h2">{props.recipe.name}</h2>
      <img className="fit" src={props.recipe.image} alt={props.recipe.image} />
      <div>
        <span>{props.recipe.category}</span>
        <span>{props.recipe.calories}</span>
        <h3>Ingredients</h3>
        {props.recipe.ingredients && (
          <ul>
            {props.recipe.ingredients.map(ingredient => (
              <li key={ingredient}>{ingredient}</li>
            ))}
          </ul>
        )};
        <h3>Steps</h3>
        {props.recipe.steps && (
          <ol>{props.recipe.steps.map(step => <li key={step}>{step}</li>)}</ol>
        )}
        <Link
          to={`/recipe/${props.recipe.id}`}
          href={`/recipe/${props.recipe.id}`}
        >
          Read More
        </Link>
      </div>
    </div>
  );
};

RecipeDetail.propTypes = {
  style: PropTypes.object,
  recipe: PropTypes.object,
  className: PropTypes.string,
};
export default RecipeDetail;
