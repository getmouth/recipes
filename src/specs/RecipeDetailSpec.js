import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import RecipeDetail from '../components/RecipeDetail';

describe('<RecipeDetail />', () => {
  let testRecipe;

  beforeEach(() => {
    testRecipe = {
      id: 1,
      name: 'Test Recipe',
      category: 'Test category',
      ingredients: ['ing 1', 'ing 2'],
      steps: ['step 1', 'step 2'],
    };
  });

  test('Should render zero state', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should correctly render a recipe', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={testRecipe} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should correctly assign a className', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={testRecipe} className="class1" />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render correctly without ingredients', () => {
    delete testRecipe.ingredients;
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={testRecipe} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render correctly without steps', () => {
    delete testRecipe.steps;
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={testRecipe} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
