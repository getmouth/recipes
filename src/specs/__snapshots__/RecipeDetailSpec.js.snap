// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeDetail /> Should correctly assign a className 1`] = `
<div
  className="p2 bg-white class1"
  style={undefined}
>
  <h2
    className="h2"
  >
    Test Recipe
  </h2>
  <img
    alt={undefined}
    className="fit"
    src={undefined}
  />
  <div>
    <span>
      Test category
    </span>
    <span />
    <h3>
      Ingredients
    </h3>
    <ul>
      <li>
        ing 1
      </li>
      <li>
        ing 2
      </li>
    </ul>
    ;
    <h3>
      Steps
    </h3>
    <ol>
      <li>
        step 1
      </li>
      <li>
        step 2
      </li>
    </ol>
    <a
      href="/recipe/1"
      onClick={[Function]}
    >
      Read More
    </a>
  </div>
</div>
`;

exports[`<RecipeDetail /> Should correctly render a recipe 1`] = `
<div
  className="p2 bg-white"
  style={undefined}
>
  <h2
    className="h2"
  >
    Test Recipe
  </h2>
  <img
    alt={undefined}
    className="fit"
    src={undefined}
  />
  <div>
    <span>
      Test category
    </span>
    <span />
    <h3>
      Ingredients
    </h3>
    <ul>
      <li>
        ing 1
      </li>
      <li>
        ing 2
      </li>
    </ul>
    ;
    <h3>
      Steps
    </h3>
    <ol>
      <li>
        step 1
      </li>
      <li>
        step 2
      </li>
    </ol>
    <a
      href="/recipe/1"
      onClick={[Function]}
    >
      Read More
    </a>
  </div>
</div>
`;

exports[`<RecipeDetail /> Should render correctly without ingredients 1`] = `
<div
  className="p2 bg-white"
  style={undefined}
>
  <h2
    className="h2"
  >
    Test Recipe
  </h2>
  <img
    alt={undefined}
    className="fit"
    src={undefined}
  />
  <div>
    <span>
      Test category
    </span>
    <span />
    <h3>
      Ingredients
    </h3>
    ;
    <h3>
      Steps
    </h3>
    <ol>
      <li>
        step 1
      </li>
      <li>
        step 2
      </li>
    </ol>
    <a
      href="/recipe/1"
      onClick={[Function]}
    >
      Read More
    </a>
  </div>
</div>
`;

exports[`<RecipeDetail /> Should render correctly without steps 1`] = `
<div
  className="p2 bg-white"
  style={undefined}
>
  <h2
    className="h2"
  >
    Test Recipe
  </h2>
  <img
    alt={undefined}
    className="fit"
    src={undefined}
  />
  <div>
    <span>
      Test category
    </span>
    <span />
    <h3>
      Ingredients
    </h3>
    <ul>
      <li>
        ing 1
      </li>
      <li>
        ing 2
      </li>
    </ul>
    ;
    <h3>
      Steps
    </h3>
    <a
      href="/recipe/1"
      onClick={[Function]}
    >
      Read More
    </a>
  </div>
</div>
`;

exports[`<RecipeDetail /> Should render zero state 1`] = `
<p
  className="h3 p2 bg-white italic center"
  style={undefined}
>
  Please select a recipe to view the details
</p>
`;
